bcmc/src/main/java/com/adyen/checkout/bcmc/DefaultBcmcDelegate.kt

```diff
@@ -126,6 +133,46 @@ internal class DefaultBcmcDelegate(
         }
     }
 
+    @Suppress("TooGenericExceptionCaught")
+    private fun fetchDelegatedAuthenticationData(coroutineScope: CoroutineScope) {
+        Logger.d(TAG, "fetchDelegatedAuthenticationData")
+        coroutineScope.launch {
+            delegatedAuthenticationData = try {
+                val availabilityResult = AdyenAuthentication.checkAvailability(application)
+                val sdkOutput = (availabilityResult as? AvailabilityResult.Available)?.sdkOutput
+                sdkOutput?.let {
+                    DelegatedAuthenticationData(it)
+                }
+            } catch (e: ClassNotFoundException) {
+                Logger.e(
+                    TAG,
+                    "delegatedAuthenticationData not set because Authentication SDK is not present in project."
+                )
+                null
+            } catch (e: NoClassDefFoundError) {
+                Logger.e(
+                    TAG,
+                    "delegatedAuthenticationData not set because Authentication SDK is not present in project."
+                )
+                null
+            } catch (e: IncompatibleClassChangeError) {
+                Logger.e(
+                    TAG,
+                    "delegatedAuthenticationData not set because " +
+                        "Authentication SDK version is incompatible with compiled version."
+                )
+                null
+            } catch (e: NullPointerException) {
```

Comment:
Can't we make it work for tests? I don't like having code just for tests 😬

Reply:
Spend some time and found a solution. I've mocked the retrieving part in the tests and removed this catch 👍